src/main/java/seedu/address/logic/commands/task/SearchTaskCommand.java

```diff
@@ -0,0 +1,88 @@
+package seedu.address.logic.commands.task;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+import static seedu.address.logic.parser.task.TaskCliSyntax.PREFIX_DUE_DATE;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import javafx.collections.ObservableList;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.room.Room;
+import seedu.address.model.task.DateTimeDue;
+import seedu.address.model.task.Task;
+
+/**
+ * Search a {@code Task} before a {@code DateTimeDue}.
+ */
+public class SearchTaskCommand extends Command {
+
+    public static final String COMMAND_WORD = "searchtask";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD + ": Searches tasks in Covigent. "
+            + "Parameters: "
+            + "[" + PREFIX_DUE_DATE + "DUE DATE]\n"
+            + "Example: " + COMMAND_WORD + " "
+            + PREFIX_DUE_DATE + "20200928 2359";
+
+    public static final String MESSAGE_SEARCH_TASK_SUCCESS = "Tasks before the due date found: \n";
+    public static final String MESSAGE_TASK_NOT_FOUND = "There is no task that matches your criteria.";
+
+
+    private final DateTimeDue duedate;
+    private ObservableList<Task> tasks;
+
+    /**
+     * Creates an SearchTaskCommand to Search the specified {@code Task} with {@code DateTimeDue}
+     */
+    public SearchTaskCommand(DateTimeDue duedate) {
+        requireAllNonNull(duedate);
+        this.duedate = duedate;
+    }
+
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        requireNonNull(model);
+        List<Room> rooms = model.getRoomList();
+        ArrayList<Task> taskListWithDesirableResult = new ArrayList<>();
+        for (Room room : rooms) {
+            tasks = room.getTaskList().asUnmodifiableObservableList();
+            for (Task task : tasks) {
+                if (task.getDueAt().compareTo(duedate) == 1) {
+                    taskListWithDesirableResult.add(task);
+                }
+            }
+        }
+
+        if (taskListWithDesirableResult.size() < 1) {
+            throw new CommandException(MESSAGE_TASK_NOT_FOUND);
+        }
+
+        return new CommandResult(String.format(MESSAGE_SEARCH_TASK_SUCCESS
+                + getListOutput(taskListWithDesirableResult)));
+    }
+
+    /**
+     * Return the list of tasks' details
```

Comment:
can consider `Returns` instead of `Return`? also, do we have to leave a line after line 70? i kind of remembered that it is in the coding convention but im not sure.

Reply:
I am not sure whether we need to leave a line also, but thanks for the "s" reminder!